temporalio/ext/src/worker.rs

```diff
@@ -0,0 +1,407 @@
+            activity,
+            _workflow,
+        })
+    }
+
+    pub fn async_poll_all(workers: RArray, queue: Value) -> Result<(), Error> {
+        // Get the first runtime handle
+        let runtime = workers
+            .entry::<typed_data::Obj<Worker>>(0)?
+            .runtime_handle
+            .clone();
+
+        // Create stream of poll calls
+        // TODO(cretz): Map for workflow pollers too
+        let worker_streams = workers
+            .into_iter()
+            .enumerate()
+            .filter_map(|(index, worker_val)| {
+                let worker_typed_data = RTypedData::from_value(worker_val).expect("Not typed data");
+                let worker_ref = worker_typed_data.get::<Worker>().expect("Not worker");
+                if worker_ref.activity {
+                    let worker = Some(
+                        worker_ref
+                            .core
+                            .borrow()
+                            .as_ref()
+                            .expect("Unable to borrow")
+                            .clone(),
+                    );
+                    Some(Box::pin(stream::unfold(worker, move |worker| async move {
+                        // We return no worker so the next streamed item closes
+                        // the stream with a None
+                        if let Some(worker) = worker {
+                            let res =
+                                temporal_sdk_core_api::Worker::poll_activity_task(&*worker).await;
+                            let shutdown_next = matches!(res, Err(PollActivityError::ShutDown));
+                            Some((
+                                (index, WorkerType::Activity, res),
+                                // No more worker if shutdown
+                                if shutdown_next { None } else { Some(worker) },
+                            ))
+                        } else {
+                            None
+                        }
+                    })))
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
+        let mut worker_stream = stream::select_all(worker_streams);
+
+        // Continually call the callback with the worker and the result. The result can either be:
+        // * [worker index, :activity/:workflow, bytes] - poll success
+        // * [worker index, :activity/:workflow, error] - poll fail
+        // * [worker index, :activity/:workflow, nil] - worker shutdown
+        // * [nil, nil, nil] - all pollers done
+        let callback = Arc::new(AsyncCallback::from_queue(queue));
+        let complete_callback = callback.clone();
+        let async_command_tx = runtime.async_command_tx.clone();
+        runtime.spawn(
+            async move {
+                // Get next item from the stream
+                while let Some((worker, worker_type, result)) = worker_stream.next().await {
```

Comment:
Maybe call worker `worker_ix`. Slightly confused me on first read.

Reply:
Renamed to `worker_index`